src/string.cr

```diff
@@ -2402,6 +2402,17 @@ class String
     true
   end
 
+  # Returns `self` if not blank, otherwise returns `nil`.
+  #
+  # ```
+  # "".presence        # => nil
+  # "   ".presence     # => nil
+  # "   a   ".presence # => "   a   "
```

Comment:
maybe we should add an example like 

```crystal
"a".try(&.presence) || "default" # => "a"
"".try(&.presence) || "default"  # => "default"
nil.try(&.presence) || "default" # => "default"
```

That would be a common use case unless `Nil#presence` is added.

Reply:
Let's add `Nil#presence`

Reply:
> Let's add `Nil#presence`

You mean to add `String#presence` and `Nil.presence`?

Reply:
No, `Nil#presence` so the following compiles:
```cr
data = {} of Symbol => String
value = data[:foo]?.presence || "bar"

Reply:
I made a mistake, of course I meant `Nil#presence`. Now it makes sense, `Nil#presence` and `String#presence` will allow to compile the code above.

Reply:
shinny sample snippets:

```cr
"a".presence || "default" # => "a"
"".presence || "default"  # => "default"
nil.presence || "default" # => "default"

data = {} of Symbol => String
data[:missing_key]?.presence || "default" # => "default"
```

Reply:
Looks great! I will update PR a bit later.

Reply:
Hmm, seems like we slowly drift back onto the #7824 turf... ;)

Reply:
Indeed. I think these methods are good, but they are mostly useful for web frameworks where the strings come from query parameters that might or might not be there. Or from form parameters. So I still wonder where these should be included in the standard library. The std is not meant to be thought exclusively for web apps.

Reply:
@asterite I guess it's useful for general input parsing, not only web. Say configuration files or so. 

But then... I've never really needed `String#presence` in Crystal. Which might mean it's not too important to have it?

Reply:
@straight-shoota @asterite I found that I need this method during CSV file parsing, so this is not only a web-needed method.

Reply:
@asterite `blank?` and `presence` come from Rails so we think of them as mostly useful for web development, but they're actually nice to have whenever we deal with user input.

A use case, for example, is dealing with legacy databases where many columns are `""` or `NULL` and should be treated the same, `#presence` is a life savior.

Reply:
I think that as long as we don't put presence in `Object` then we are slightly better than #7824.